packages/nvidia_nat_core/src/nat/observability/exporter_manager.py

```
@@ -202,59 +204,61 @@ async def start(self, context_state: ContextState | None = None):
-            # Then stop the manager tasks
-            await self.stop()
+            async with self._lock:
+                self._ref_count -= 1
+                should_stop = self._ref_count == 0
+
+            if should_stop:
+                # Last caller — actually shut down the exporters
+                try:
+                    await self._cleanup_isolated_exporters()
+                except Exception as e:
+                    logger.exception("Error during isolated exporter cleanup: %s", e)
```

Ruff (0.15.0)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)